Frends.SFTP.UploadFiles/Frends.SFTP.UploadFiles/Definitions/SingleFileTransfer.cs

```diff
@@ -1,488 +1,464 @@
 ﻿using System.Text;
 using Renci.SshNet;
-using Renci.SshNet.Sftp;
 
-namespace Frends.SFTP.UploadFiles.Definitions
+namespace Frends.SFTP.UploadFiles.Definitions;
+
+internal class SingleFileTransfer
 {
-    /// <summary>
-    /// Class for executing single file transfer.
-    /// </summary>
-    internal class SingleFileTransfer
-    {
-        private readonly RenamingPolicy _renamingPolicy;
-        private readonly ISFTPLogger _logger;
-        private SingleFileTransferResult _result;
+    private readonly RenamingPolicy _renamingPolicy;
+    private readonly ISFTPLogger _logger;
+    private readonly SingleFileTransferResult _result;
 
-        public SingleFileTransfer(FileItem file, BatchContext context, SftpClient client, RenamingPolicy renamingPolicy, ISFTPLogger logger)
-        {
-            _renamingPolicy = renamingPolicy;
-            _logger = logger;
+    public SingleFileTransfer(FileItem file, string destinationDirectory, BatchContext context, SftpClient client, RenamingPolicy renamingPolicy, ISFTPLogger logger)
+    {
+        _renamingPolicy = renamingPolicy;
+        _logger = logger;
 
-            FileTransferStartTime = DateTime.UtcNow;
-            Client = client;
-            SourceFile = file;
-            BatchContext = context;
+        FileTransferStartTime = DateTime.UtcNow;
+        Client = client;
+        SourceFile = file;
+        BatchContext = context;
 
-            DestinationFileNameWithMacrosExpanded = renamingPolicy.CreateRemoteFileName(
-                    file.Name,
-                    context.Destination.FileName);
+        DestinationFileWithMacrosExpanded = Path.Combine(destinationDirectory, renamingPolicy.CreateRemoteFileName(
+                file.Name,
+                context.Destination.FileName));
 
-            _result = new SingleFileTransferResult { Success = true };
-        }
+        _result = new SingleFileTransferResult { Success = true };
+    }
 
-        public DateTime FileTransferStartTime { get; set; }
+    public DateTime FileTransferStartTime { get; set; }
 
-        public SftpClient Client { get; set; }
-        public FileItem SourceFile { get; set; }
-        public FileItem DestinationFile { get; set; }
-        public string DestinationFileNameWithMacrosExpanded { get; set; }
-        private string SourceFileDuringTransfer { get; set; }
-        public string DestinationFileDuringTransfer { get; set; }
-        internal BatchContext BatchContext { get; set; }
+    public SftpClient Client { get; set; }
+    public FileItem SourceFile { get; set; }
+    public FileItem DestinationFile { get; set; }
```

Comment:
You are not using this property, so it can be removed.

Reply:
Removed